### PR TITLE
[SDK-2029] Run framework integration automatically

### PR DESCRIPTION
**Pull request by frederikprijck**
**Github Actions:**
- Added 4 checks, one for each framework
- Each check runs in parralel and isolated
- Every step builds the SPAJS SDK
- Every step creates a new application for the framework in question
- Every step imports the Auth0Client and creates a new instance.
- Every step builds the generated application to verify it can build using our SPA.

> **Edit**: Rolled back the above changes as it seems likely to go with Circle CI.

**Circle CI:**
- Added 4 jobs, one for each framework
  - Every step builds the SPAJS SDK
  - Every step creates a new application for the framework in question
  - Every step imports the Auth0Client and creates a new instance.
  - Every step builds the generated application to verify it can build using our SPA.
- Added an extra job to extract the integration tests from the build job
- Added workflow support to control job execution dependencies.

### CODE REVIEW

**Comment by stevehobbsdev** [comment]
@frederikprijck Are you able to remove the reviewer and mark the PR as a draft? Will the action still run?

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,3 +32,154 @@ jobs:
           path: cypress/screenshots
       - store_artifacts:
           path: cypress/videos
+  test_gatsby:
+    docker:
+      - image: cypress/browsers:node12.13.0-chrome80-ff74
```
Comment (frederikprijck) [inline .circleci/config.yml:37]: I used the same image as we use for the integration tests. I would argue this is not needed, but I got permission error on some others I tried (node alpine, ubuntu, ...), so I figured sticking to the one we are already using wouldn't hurt. What do u think?

**Review by stevehobbsdev — COMMENTED** [review]
🎉  Looks great. I have a feeling we're going to run with the Circle version, so if that's the case remember to remove the GHA workflow first.

File: .circleci/config.yml (hunk 1 above)
Comment (frederikprijck) [reply .circleci/config.yml:37]: Good call, I will have a look!

**Comment by frederikprijck** [comment]
Reverted the GitHub Actions commit.

**Review by stevehobbsdev — DISMISSED** [review]

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**